cpp/src/parquet/column_reader.cc

```diff
@@ -1347,16 +1346,24 @@ class TypedRecordReader : public TypedColumnReaderImpl<DType>,
     int64_t gap = levels_position_ - start_levels_position;
     if (gap == 0) return;
 
-    std::copy(def_levels() + levels_position_, def_levels() + levels_written_,
-              def_levels() + levels_position_ - gap);
+    int64_t levels_remaining = levels_written_ - gap;
+
+    int16_t* def_data = def_levels();
+    std::copy(def_data + levels_position_, def_data + levels_written_,
+              def_data + levels_position_ - gap);
+    PARQUET_THROW_NOT_OK(def_levels_->Resize(levels_remaining * sizeof(int16_t), false));
```

Comment:
nit: comment literal.  This doesn't try to release the memory correct?

Reply:
Correct. Added parameter comments.